### PR TITLE
Generates more models from the API specs

### PR DESCRIPTION
**Pull request by cloudify**
CreatedMessage, CreatedMessageResponse, ExtendedProfile, LimitedProfile

### CODE REVIEW

File: api/public_api_v1.yaml [hunk 1]
```diff
@@ -104,17 +91,18 @@ paths:
...
           schema:
             $ref: '#/definitions/ProblemJson'
-  '/preferences/{fiscal_code}':
-    x-swagger-router-controller: Preferences
+  '/profiles/{fiscal_code}':
     get:
-      operationId: getUserPreferences
+      operationId: getProfile
       summary: Get User's Preferences
       description: Returns the preferences for the user identified by the provided fiscal code.
```
Comment (gunzip) [inline api/public_api_v1.yaml]: align description with the new 'profile' naming. 
moreover the description should explain when it returns Limited vs Extended profile.

File: lib/controllers/messages.ts [hunk 2]
```diff
@@ -308,10 +298,15 @@ export function CreateMessageHandler(
...
 
+      const responsePayload: CreatedMessageResponse = {
+        dry_run: false
+      };
+
       // redirect the client to the message resource
       return ResponseSuccessRedirectToResource(
         retrievedMessage,
-        `/api/v1/messages/${fiscalCode}/${message.id}`
+        `/api/v1/messages/${fiscalCode}/${message.id}`,
```
Comment (gunzip) [inline lib/controllers/messages.ts]: better to not hardcode API paths (expecially true for version)

File: lib/utils/response.ts [hunk 3]
```diff
@@ -78,21 +78,24 @@ export function ResponseSuccessJsonIterator<T>(
...
+export function ResponseSuccessRedirectToResource<T, V>(
   resource: T,
-  url: string
-): IResponseSuccessRedirectToResource<T> {
+  url: string,
+  payload: V
+): IResponseSuccessRedirectToResource<T, V> {
   return {
-    apply: res => res.redirect(202, url),
+    apply: res => res.json(payload).redirect(202, url),
```
Comment (gunzip) [inline lib/utils/response.ts]: why ?

File: api/public_api_v1.yaml (hunk 1 above)
Comment (cloudify) [reply api/public_api_v1.yaml]: descriptions and operations will be my next effort (contribution welcome), I'm currently focusing on generating models

File: lib/controllers/messages.ts (hunk 2 above)
Comment (cloudify) [reply lib/controllers/messages.ts]: they have to be hardcoded _somewhere_ :)

File: lib/utils/response.ts (hunk 3 above)
Comment (cloudify) [reply lib/utils/response.ts]: for consistency, I'd like to always return a response :)

File: lib/controllers/messages.ts (hunk 2 above)
Comment (cloudify) [reply lib/controllers/messages.ts]: also that code is not part of this PR, I'll add a reminder in the backlog

Comment (cloudify) [reply lib/controllers/messages.ts]: https://www.pivotaltracker.com/story/show/151217639

**Pull request merged**